Branch-SDK/src/io/branch/referral/SystemObserver.java

```diff
@@ -429,32 +445,39 @@ static String getLocalIPAddress() {
      * {#UI_MODE_TYPE_WATCH Configuration.UI_MODE_TYPE_WATCH}.
      */
     String getUIMode() {
-        String mode;
-        switch (((UiModeManager) context_.getSystemService(UI_MODE_SERVICE)).getCurrentModeType()) {
-            case 0:
-                mode = "UI_MODE_TYPE_UNDEFINED";
-                break;
-            case 1:
-                mode = "UI_MODE_TYPE_NORMAL";
-                break;
-            case 2:
-                mode = "UI_MODE_TYPE_DESK";
-                break;
-            case 3:
-                mode = "UI_MODE_TYPE_CAR";
-                break;
-            case 4:
-                mode = "UI_MODE_TYPE_TELEVISION";
-                break;
-            case 5:
-                mode = "UI_MODE_TYPE_APPLIANCE";
-                break;
-            case 6:
-                mode = "UI_MODE_TYPE_WATCH";
-                break;
-            default:
-                mode = "UI_MODE_TYPE_UNDEFINED";
-                break;
+        String mode = "UI_MODE_TYPE_UNDEFINED";
+        Context context = contextReference_.get();
+        UiModeManager modeManager = null;
+
+        if (context != null) {
+            modeManager = (UiModeManager)context.getSystemService(UI_MODE_SERVICE);
```

Comment:
Feel like we should throw a try catch around this statement

Reply:
I can do this, but the change here was to separate out the Context (which seemed in the stack trace to be "dead") and the getting of the system service (which can return null) and finally the calling of that method.

To be really safe I can wrap the entire method, but we shouldn't be guessing here on what the original problem was that caused the exception -- that might just push the problem further down the path.

Let me know if you want me to wrap just the code indicated, or the whole method.

Reply:
Reverted Weak Reference and wrapped this method with a try/catch